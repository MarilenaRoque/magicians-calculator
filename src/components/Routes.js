import { BrowserRouter, Switch, Route } from 'react-router-dom';
import React from 'react';
import Calculator from './Calculator';
import Home from './Home';
import Quote from './Quote';
import Navbar from './Navbar';

export default function Routes() {
  return (
    <BrowserRouter>
      <Navbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/Calculator" component={Calculator} />
        <Route exact path="/Quote" component={Quote} />
      </Switch>
    </BrowserRouter>
  );
}
