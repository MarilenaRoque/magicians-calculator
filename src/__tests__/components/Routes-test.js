import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Routes from '../../components/Routes';
import '@testing-library/jest-dom';

describe('Routes Component', () => {
  const tree = renderer.create(<Routes />).toJSON();
  it('renders Routes page correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
describe('App content and Navbar links', () => {
  it('Default content page is Home', () => {
    render(<Routes />);
    expect(screen.getByTestId('page-content')).toHaveTextContent('Welcome to Math Magicians!');
  });
  it('Calculator Page is displayed if you click in the Calculator Link in the Navbar', () => {
    render(<Routes />);
    fireEvent.click(screen.getByText('Calculator'));
    expect(screen.getByTestId('page-content')).toHaveTextContent('Let’s do some math!');
  });
  it('Calculator Page is displayed if you click in the Calculator Link in the Navbar', () => {
    render(<Routes />);
    fireEvent.click(screen.getByText('Quote'));
    expect(screen.getByTestId('page-content')).toHaveTextContent('You’re pointless.');
  });
});
