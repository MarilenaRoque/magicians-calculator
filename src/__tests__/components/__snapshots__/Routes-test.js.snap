// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component renders Routes page correctly 1`] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "15px",
      }
    }
  >
    <img
      alt="Math Magicians Logo"
      src="logo-math.svg"
      style={
        Object {
          "height": "100px",
        }
      }
    />
    <div
      className="right"
      style={
        Object {
          "fontSize": "28px",
        }
      }
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "28px",
          }
        }
      >
         Home 
      </a>
       
      |
      <a
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "28px",
          }
        }
      >
         Calculator 
      </a>
        
      |
      <a
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "28px",
          }
        }
      >
         Quote 
      </a>
        
    </div>
  </div>,
  <div
    data-testid="page-content"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": "auto",
          "width": "70%",
        }
      }
    >
      <h2
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
        Welcome to Math Magicians!
      </h2>
      <p
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
        Math magicians is the tool that you needed to solve your mathematical problems in an easy, faster and fun way. We want to help you and guide you through the Magic Math Land and make you explore the best mathematical experiences.
      </p>
      <p
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
        Okay...I admit that was corny, but we all know math isnâ€™t always the most exciting subject to teach. That means you have to find strategies to make lessons fun, like gamification in the classroom, math puzzles or -- in this case -- math jokes that will lighten the mood and brighten the vibe in your classroom.
      </p>
    </div>
  </div>,
]
`;
